.github/workflows/release.yml

```diff
@@ -46,37 +24,38 @@ jobs:
           ruby-version: 2.7
           bundler-cache: true
 
-      - name: "build"
+      - name: "get version"
+        id: "get_version"
+        run: |
+          echo "::set-output name=version::$(jq --raw-output .version metadata.json)"
+
+      - name: "build gem"
         run: |
           bundle exec rake build
 
-      - name: "publish"
+      - name: "publish gem"
         run: |
-          bundle exec rake push
+          gem push ./pkg/*.gem
```

Comment:
maybe interesting for you: At Vox Pupuli we wanted to avoid a vendor lockin/single out of failure (we once opted in for travis and migrating away was a huge pain). Because of that, we now publish gems to rubygems.org and to GitHub:
https://github.com/voxpupuli/beaker/blob/master/.github/workflows/release.yml

Reply:
I've noticed this and really like the idea. Its something I'd like to add to our reusable workflows.